Non durable transport documentation and sample (#8437)
Particular/docs.particular.net: transports/non-durable/index.md

- High throughput is required and the overhead of a persistent queueing system is unnecessary.
- Starting with synchronous, mediator-like behavior and gradually moving toward true asynchronicity.
- Integrating with webhooks: the HTTP endpoint immediately acknowledges the caller by dispatching work to a local non-durable queue. NServiceBus retries the work as needed, and transient failures can be discarded rather than moved to an error queue.

> [!WARNING]
> Messages are lost when the process ends. Ensure this trade-off is acceptable for the target scenario.

## Transport at a glance

|Feature                    |   |
|:---                       |---
|Transactions               |None, ReceiveOnly, SendsAtomicWithReceive
|Pub/Sub                    |Native
|Timeouts                   |Native
|Large message bodies       |In-memory only; limited by available process memory
|Scale-out                  |Competing consumer (single process only)
|Scripted Deployment        |Not supported
|Installers                 |Not supported
|Native integration         |Not supported
|OpenTelemetry tracing      |[Supported](observability.md)
|Case Sensitive             |Yes

> |Aspire integration         |Not yet supported

Author: ```suggestion
|Aspire integration         |No
```

Use both together when:

- Sagas must be updated atomically with message dispatch.
- A single-process system needs transactional message handling without external infrastructure.
- The overhead of a persistent outbox or distributed transaction coordinator is undesirable.

## Inline execution

The transport supports an optional inline execution mode. When enabled, sends to local queues are executed synchronously in the sending thread rather than being enqueued for the pump to process later. This behaves like a mediator pattern: the caller waits for the handler to complete and receives any exceptions immediately.

This is useful for:

- Gradually moving a codebase from synchronous method calls toward asynchronous messaging.
- Scenarios that require mediator-like behavior with immediate feedback.
- Tests that need deterministic, immediate execution without background polling.

> snippet: NonDurableTransport-InlineExecution

Reviewer: ```suggestion
Providing a non-null `InlineExecutionOptions` enables the inline execution mode, and additional settings are available as properties on the class:

snippet: NonDurableTransport-InlineExecution
```

An inline operation attempted after its destination receiver begins stopping is rejected. The originating parent message then follows its configured recoverability policy. If recoverability requests a retry, the parent message is requeued and remains buffered until processing restarts.

If a cancellation token is provided to the Stop method and it signals cancellation, the in-flight message handlers will be interrupted to force the endpoint to stop faster.

Buffered messages remain in the queue on shutdown. They can be processed if the same receiver starts again, for example, through `ChangeConcurrency`; otherwise they are lost when the `NonDurableBroker` is disposed. A new endpoint using the same broker can process a buffered message, but it cannot complete an inline-execution dispatch task owned by the previous endpoint instance. Use `DrainQueueBeforeShutdown` if inline cascades must be given an opportunity to complete before shutdown returns.

> ### Broker disposal

Reviewer: This feels like it should be a top-level heading?

```suggestion
## Broker disposal
```

- Endpoints run in the same process and data loss on restart is acceptable.
- High throughput is required and the overhead of a persistent queueing system is unnecessary.
- Starting with synchronous, mediator-like behavior and gradually moving toward true asynchronicity.
- Integrating with webhooks: the HTTP endpoint immediately acknowledges the caller by dispatching work to a local non-durable queue. NServiceBus retries the work as needed, and transient failures can be discarded rather than moved to an error queue.

> [!WARNING]
> Messages are lost when the process ends. Ensure this trade-off is acceptable for the target scenario.

## Transport at a glance

|Feature                    |   |
|:---                       |---
|Transactions               |None, ReceiveOnly, SendsAtomicWithReceive
|Pub/Sub                    |Native
|Timeouts                   |Native
|Large message bodies       |In-memory only; limited by available process memory
|Scale-out                  |Competing consumer (single process only)
|Scripted Deployment        |Not supported
|Installers                 |Not supported
|Native integration         |Not supported
|Case Sensitive             |Yes

> |Aspire integration         |Not yet supported

Reviewer: Bookmark: another mention of Aspire (refer to other comment)

## Prerequisites

> - All endpoints must exist in the same process.

Reviewer: ```suggestion
- All endpoints must be hosted in the same process.
```
